### ISSUE TITLE
error[E0700]: hidden type for `impl Trait` captures lifetime that does not appear in bounds

### ISSUE THREAD

**Issue by MOZGIII** [dtolnay/async-trait#15]
With this seemingly ok code:

```rust
#![feature(async_await)]

use async_trait::async_trait;
use slog::Logger;

#[async_trait]
trait MyTrait {
    async fn myfn(&self, logger: Logger);
}

struct MyImpl;

#[async_trait]
impl MyTrait for MyImpl {
    async fn myfn(&self, _logger: Logger) {}
}
```

I get this odd error:

```
   Compiling redacted v0.1.0 (/redacted)
error[E0700]: hidden type for `impl Trait` captures lifetime that does not appear in bounds
  --> tests/test.rs:15:43
   |
15 |     async fn myfn(&self, _logger: Logger) {}
   |                                           ^
   |
note: hidden type `impl std::future::Future` captures the scope of call-site for function at 15:43
  --> tests/test.rs:15:43
   |
15 |     async fn myfn(&self, _logger: Logger) {}
   |                                           ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0700`.
error: Could not compile `redacted`.

To learn more, run the command again with --verbose.
```

The error is caused by `slog::Logger`.

This seems to be a bug. As a workaround, I'm currently passing the logger in a wrapper-struct, but I hope to get this solved.

PS: I have no idea what's going on here, and how to simplify the example further.

**Comment by dtolnay** [comment]
Thanks! I filed this as a compiler bug: https://github.com/rust-lang/rust/issues/63033. I will try to implement a workaround.

**Comment by dtolnay** [comment]
Fixed in 0.1.6.
